applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py

```diff
@@ -298,12 +298,24 @@ def InitializeSolutionStep(self):
             if (self.settings["formulation"]["element_type"].GetString() == "embedded_ausas_navier_stokes"):
                 (self.find_nodal_neighbours_process).Execute()
 
-            # Perform the FM-ALE operations
-            self._do_fm_ale_operations()
+            # Set the virtual mesh values from the background mesh
+            self._set_virtual_mesh_values()
 
             # Fluid solver step initialization
             (self.solver).InitializeSolutionStep()
 
+    def SolveSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            # Perform the FM-ALE operations
+            self._do_fm_ale_operations()
+
+            # Solve the embedded CFD problem
+            is_converged = self.solver.SolveSolutionStep()
+            if not is_converged and self._IsPrintingRank():
+                msg  = "Fluid solver did not converge for step " + str(self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]) + "\n"
+                msg += "corresponding to time " + str(self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]) + "\n"
+                KratosMultiphysics.Logger.PrintWarning("FluidSolver",msg)
+
```

Comment:
I suggest the following:
~~~py
if self._TimeBufferIsInitialized():
    # Perform the FM-ALE operations
    self._do_fm_ale_operations()

    super.SolveSolutionStep()
~~~

Ofc `SolveSolutionStep `will again check if the buffer is initialized, but I think this a worthy sacrifice for avoiding copy-pasting of code

Reply:
I did the same reasoning you did. Let's do it in your way. I'd be better in case we change the base `SolveSolutionStep()` in the future.